ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/video/VideoComponentState.kt

```diff
@@ -0,0 +1,344 @@
+                density = density,
+            ),
+            height = height.adjustDimension(
+                other = width,
+                thisDimensionPx = videoUrls.height,
+                otherDimensionPx = videoUrls.width,
+                density = density,
+            ),
+        )
+
+    /**
+     * Adjusts this size constraint to take into account the size of the video.
+     */
+    private fun SizeConstraint.adjustDimension(
+        other: SizeConstraint,
+        thisDimensionPx: UInt,
+        otherDimensionPx: UInt,
+        density: Density,
+    ): SizeConstraint = when (this) {
+        is Fit -> {
+            when (other) {
+                is Fit -> Fixed(with(density) { thisDimensionPx.toInt().toDp().value.toUInt() })
+                is Fill -> this
+
+                is Fixed -> {
+                    // If the other dimension is Fixed, we'll have to scale this one by the same factor.
+                    val otherDimensionDp = with(density) { otherDimensionPx.toInt().toDp() }
+                    val scaleFactor = other.value.toFloat() / otherDimensionDp.value
+                    Fixed(with(density) { (scaleFactor * thisDimensionPx.toInt()).toDp().value.toUInt() })
+                }
+            }
+        }
+
+        is Fill,
+        is Fixed,
+        -> this
+    }
+}
+
+@Stable
+@JvmSynthetic
+@Composable
+internal fun rememberUpdatedVideoComponentState(
+    style: VideoComponentStyle,
+    paywallState: PaywallState.Loaded.Components,
+): VideoComponentState =
+    rememberUpdatedVideoComponentState(
+        style = style,
+        localeProvider = { paywallState.locale },
+        selectedPackageProvider = { paywallState.selectedPackageInfo?.rcPackage },
+        selectedTabIndexProvider = { paywallState.selectedTabIndex },
+    )
+
+@Stable
+@JvmSynthetic
+@Composable
+internal fun rememberUpdatedVideoComponentState(
+    style: VideoComponentStyle,
+    localeProvider: () -> Locale,
+    selectedPackageProvider: () -> Package?,
+    selectedTabIndexProvider: () -> Int,
+): VideoComponentState {
+    val windowSize = currentWindowAdaptiveInfo().windowSizeClass.windowWidthSizeClass
+    val density = LocalDensity.current
+    val darkMode = isSystemInDarkTheme()
+    val layoutDirection = LocalLayoutDirection.current
+
+    return remember(style) {
+        VideoComponentState(
+            initialWindowSize = windowSize,
+            initialDensity = density,
+            initialDarkMode = darkMode,
+            initialLayoutDirection = layoutDirection,
+            style = style,
+            localeProvider = localeProvider,
+            selectedPackageProvider = selectedPackageProvider,
+            selectedTabIndexProvider = selectedTabIndexProvider,
+        )
+    }.apply {
+        update(
+            windowSize = windowSize,
+            density = density,
+            darkMode = darkMode,
+            layoutDirection = layoutDirection,
+        )
+    }
```

Comment:
I’m curious why we’re not passing all these configurations into remember as parameters so the state can update dynamically.

```kotlin
remember(keys = style, windoeSize, densitry, darkMode, layoutDirection, etc)
```

Then I believe we can just remove the `update` function and the overused `derivedStateOf`.